src/pallene/parser.lua

```diff
@@ -285,7 +285,7 @@ function Parser:Program()
                 exp.var._tag == "ast.Var.Name" and
                 exp.var.name == modname)
             then
-                -- The checker also needs to check that this name has not been shadowed
+                -- The type checker also needs to check that this name has not been shadowed
```

Comment:
I wasn't sure about the meaning of this comment, and if it relates to typechecker or if something else was meant by “checker”?

Reply:
Good point. If we want to be picky, Scope Checking would be separate idea from Type Checking. But if you ask me it's fine to bundle it with the type checker (in our current implementation they are one and the same)